financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/internal/FinancialServicesCommonServiceComponent.java

```diff
@@ -54,4 +66,32 @@ protected void deactivate(ComponentContext context) {
 
         log.debug("Financial Services common component is deactivated");
     }
+
+    /**
+     * Initialize the connection manager for HTTPS protocol.
+     */
+    private void initConnectionManagerForHttpsProtocol() {
+
+        int maxTotal = FinancialServicesConfigParser.getInstance().getConnectionPoolMaxConnections();
+        int maxPerRoute = FinancialServicesConfigParser.getInstance().getConnectionPoolMaxConnectionsPerRoute();
+
+        try {
+            SSLConnectionSocketFactory sslsf = HTTPClientUtils.createSSLConnectionSocketFactory();
+            Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder
+                    .<ConnectionSocketFactory>create()
+                    .register(HTTP_PROTOCOL, new PlainConnectionSocketFactory())
+                    .register(HTTPS_PROTOCOL, sslsf)
+                    .build();
+
+            connectionManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+
+            connectionManager.setMaxTotal(maxTotal);
+            connectionManager.setDefaultMaxPerRoute(maxPerRoute);
+            FinancialServicesCommonDataHolder.getInstance().setConnectionManager(connectionManager);
+        } catch (FinancialServicesException e) {
+            if (connectionManager != null) {
```

Comment:
shouldn't we handle the error here?

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/514/commits/59d1abcd824194db81ac7185609cf05f6f591dd3